devicetypes/rooms-beautiful/curtain.src/curtain.groovy

```diff
@@ -0,0 +1,249 @@
+    zigbee.command(0x0102, 0x02) +
+    sendEvent(name: "replay", value: "pause")
+}
+
+private handleBatteryEvent(volts) {
+	def linkText = getLinkText(device)
+    
+    //log.warn "Value of adc: ${volts}"
+
+    if (volts > 30 || volts < 20) {
+        log.debug "${linkText} - Ignoring invalid value for voltage (${volts/10}V)"
+    }
+    else {
+        def batteryMap = [30:"full", 29:"full", 28:"full", 27:"medium", 26:"low", 25:"dead"]
+		
+        def value = batteryMap[volts]
+        if(value != null){
+            def minVolts = 25
+			def maxVolts = 30
+			def pct = (volts - minVolts) / (maxVolts - minVolts)
+			def roundedPct = Math.round(pct * 100)
+			def percent = Math.min(100, roundedPct)
+            
+        	sendEvent(name: "battery", value: percent)
+        	sendEvent(name: "battLife", value: value)
+        	log.debug "${linkText} - Batt: ${value} **** Volts: ${volts/10}v **** Percent: ${percent}%"
+        } 
+    }
+}
+
+def refresh() {
+    zigbee.onOffRefresh() +
+    zigbee.readAttribute(0x0001, 0x0020) +
+    
+    // For Diagnostics
+    zigbee.readAttribute(0xFC00, 0x0000) +
+    zigbee.readAttribute(0xDC00, 0x0000)
+}
+
+def ping() {
+    return refresh()
+}
+
+//	Don't do Device-Watch to prevent 20-30 min read attribute
+def configure() {
+    return refresh()
+}
+
+def installed() {
+	sendEvent(name: "supportedWindowShadeCommands", value: JsonOutput.toJson(["open", "close", "pause"]), displayed: true)
+	sendEvent(name: "battery", value: 100)
+	sendEvent(name: "battLife", value: "full")
+	response(refresh())
+}
+
+def updated() {
+	// Needed because updated() is being called twice
+    def time
+    if(state.updatedDate == null){
+    	time = now()
+    }
+    else{
+        time = now() - state.updatedDate
+    }
+    state.updatedDate = now()
+    
+    log.trace ("Time: ${time}")
+    
+    //	Updated [Tested on 12/27/18] - Don't need if statement
+    //	Doesn't occur twice anymore
+    
+    //if (time < 100 ){	// Smaller value (e.g. 100) means less likely to occur twice
+        if (invert.value == false)
+            response(normal())
+        else if(invert.value == true)
+            response(reverse())
+    //}
+}
+
+def normal() {
+    if(device.currentState("switch").value == "on"){
+    	sendEvent(name: "switch", value: "off")
+        log.warn ("normal-off")
+        zigbee.writeAttribute(0xFC00, 0x0000, DataType.BOOLEAN, 0x00)
+    }
+    else{
+    	sendEvent(name: "switch", value: "on")
+        log.warn ("normal-on")
+        zigbee.writeAttribute(0xFC00, 0x0000, DataType.BOOLEAN, 0x00)
+    }   
+}
+
+def reverse() {
+	if(device.currentState("switch").value == "on"){    	
+    	sendEvent(name: "switch", value: "off")
+        log.warn ("reverse-off")
+        zigbee.writeAttribute(0xFC00, 0x0000, DataType.BOOLEAN, 0x01)
+    }
+    else{
+    	sendEvent(name: "switch", value: "on")
+        log.warn ("reverse-on")
+        zigbee.writeAttribute(0xFC00, 0x0000, DataType.BOOLEAN, 0x01)
+    }
+}
```

Comment:
Do you also need to send updated windowShade events here?

Reply:
@tpmanley I still have to define the sendEvents because if the user reverses the direction of the curtain, the app should update to the correct direction. I've updated the code to sendEvent to windowShade instead of switch now.

Reply:
@devropes I'd go ahead and have the `switch` and `windowShade` events. That should keep everything happy.

Reply:
Actually, since you aren't declaring the Switch capability, then the changed code is fine as is.